lib/extensions/ar_migration.rb

```diff
@@ -0,0 +1,84 @@
+module ArPglogicalMigration
+  module PglogicalMigrationHelper
+    def self.migrations_column_present?
+      ActiveRecord::Base.connection.columns("miq_regions").any? { |c| c.name == "migrations_ran" }
+    end
+
+    def self.update_local_migrations_ran(version, direction)
+      return unless migrations_column_present?
+      return unless (region = MiqRegion.my_region)
+
+      new_migrations = ActiveRecord::SchemaMigration.normalized_versions
+      new_migrations << version if direction == :up
+      migrations_value = ActiveRecord::Base.connection.quote(PG::TextEncoder::Array.new.encode(new_migrations))
+
+      ActiveRecord::Base.connection.exec_query(<<~SQL)
+        UPDATE miq_regions
+        SET migrations_ran = #{migrations_value}
+        WHERE id = #{region.id}
+      SQL
+    end
+  end
+
+  class RemoteRegionMigrationWatcher
+    class HelperARClass < ActiveRecord::Base; end
+
+    attr_reader :region, :subscription, :version
+
+    def initialize(subscription, version)
+      @region       = MiqRegion.find_by(:region => subscription.provider_region)
```

Comment:
Here too with the MiqRegion model.

Reply:
So I was able to put together something that I think would work here, but ran into tons of issues trying to write the spec. I just went with a model stub in the class, but if this is something you feel strongly about I can probably get something working with another day's worth of work or so.